Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -219,8 +242,72 @@ public class PBXProjGenerator {
             pbxProject.projects = subprojects
         }
 
-        try project.targets.forEach(generateTarget)
-        try project.aggregateTargets.forEach(generateAggregateTarget)
+        let generateTargetsGroup = DispatchGroup()
+        var targetErrors: [Error] = []
+        var aggregateTargetsErrors: [Error] = []
+
+        var generatedTargetNames: [String] = []
+        var targetsPendingDependencyGeneration: [Target] = project.targets
+        while generatedTargetNames.count != project.targets.count {
+            var targetsReadyToGenerate = targetsPendingDependencyGeneration.filter({ [project] target in
+                let dependencies = target.getAllDependencies(usingProject: project)
+                return dependencies.reduce(into: true) { (result: inout Bool, dependency) in
+                    if dependency.reference.contains("/") {
+                        return // skip such dependencies because they are Project references most likely
+                    }
+                    result = result && generatedTargetNames.contains(dependency.reference)
```

Comment:
`generatedTargetNames.contains(dependency.reference)`, this is how we see if a target is ready to be generated

By checking each dependency of that target, and checking if it's already been generated, if so, then the target is ready to be generated, otherwise it stays in the `pendingTargets` array